kornia/x/callbacks.py

```diff
@@ -72,35 +87,50 @@ class ModelCheckpoint:
     Args:
         filepath: the where to save the mode.
         monitor: the name of the value to track.
-
+        max_mode: if true metric will be multiply by -1
+                  turn this flag when increasing metric value is expected for example Accuracy
     **Usage example:**
 
     .. code:: python
+        def my_evaluate(self) -> dict[str, AverageMeter]:
+            # stats = StatsTracker()
+            # loss = nn.CrossEntropyLoss()
+            ...
+            prediction = self.on_model(self.model, sample)
+            val_loss = self.compute_loss(out, sample["mask"])
+            stats.update("loss", val_loss.item(), batch_size)
+
+            return stats.as_dict()
 
         model_checkpoint = ModelCheckpoint(
-            filepath="./outputs", monitor="top5",
+            filepath="./outputs", monitor="loss",
         )
 
         trainer = ImageClassifierTrainer(...,
             callbacks={"on_checkpoint", model_checkpoint}
         )
     """
 
-    def __init__(self, filepath: str, monitor: str, filename_fcn: Optional[Callable[..., str]] = None) -> None:
+    def __init__(
+        self, filepath: str, monitor: str, filename_fcn: Optional[Callable[..., str]] = None, max_mode: bool = False
+    ) -> None:
         self.filepath = filepath
         self.monitor = monitor
         self._filename_fcn = filename_fcn or default_filename_fcn
-
         # track best model
-        self.best_metric: float = 0.0
+        self.best_metric: float = inf
```

Comment:
This shouldn't be equals when max mode is false?

Reply:
agree

Reply:
added -inf for max_mode and inf for min_mode with small change in code logic to be cleaner